lib/govspeak.rb

```diff
@@ -295,6 +329,19 @@ def render_image(image)
           doc.gsub!("<ul>", "<ol>")
           doc.gsub!("</ul>", "</ol>")
           doc.sub!("<ol>", '<ol class="legislative-list">')
+
+          footnotes = body.scan(/\[\^(\d)\]/)
+          counter = 0
+
+          while footnotes.size.positive? && counter < footnotes.size
```

Comment:
It would be more idiomatic to use something like `footnotes.each_with_index do |footnote, i|` and remove the `counter` variable altogether.

Reply:
Thanks @ChrisBAshton , was clearly looking at this for too long and didn't see this improvement when refactoring! 
It doesn't require the `index` as the footnote numbers can vary and exist in multiple Legislative lists, so the values are simply stored in `footnotes` - I've now updated it [here](https://github.com/alphagov/govspeak/pull/216/files#diff-acb6db6cd67285bd5380398c9e51ced8940ed7c4110706d038cbb243310e4055R349-R355)